src/sys/unix.rs

```diff
@@ -1773,6 +1773,37 @@ impl crate::Socket {
                 })
         }
     }
+
+    /// Attach Berkeley Packet Filter(BPF) on this socket.
+    ///
+    /// BPF allows a user-space program to attach a filter onto any socket
+    /// and allow or disallow certain types of data to come through the socket.
+    ///
+    /// For more information about this option, see [filter](https://www.kernel.org/doc/html/v5.12/networking/filter.html)
+    #[cfg(all(feature = "all", any(target_os = "linux", target_os = "android")))]
+    pub fn attach_filter(&self, filters: &mut [libc::sock_filter]) -> io::Result<()> {
```

Comment:
We don't need a mutable reference I think? I assume the kernel is only reading the values.

You can use `filters.as_ptr() as *mut _`.